web/client/epics/usersession.js

```diff
@@ -0,0 +1,94 @@
+ * @param {*} sessionSelector selector that builds the session object to save
+ * @param {*} idSelecotr selector of the identifier for the current session (if it already exists)
+ * @example
+ *
+ * const nameSelector = (state) => ({state.context.name + "." + state.security.user.name})
+ * const sessionSelector = (state) => ({state.map.present})
+ * const idSelector = (state) => ({state.usersession.id})
+ * const epic = saveUserSessionEpic(nameSelector, sessionSelector, idSelector)
+ * setInterval(() => dispatch({type: SAVE_USER_SESSION}), 60 * 1000)
+ */
+export const saveUserSessionEpic = (nameSelector, sessionSelector, idSelector = () => {}) => (action$, store) => action$
+    .ofType(SAVE_USER_SESSION)
+    .switchMap(() => {
+        const state = store.getState();
+        const session = sessionSelector(state);
+        const resource = {
+            category: 'USERSESSION',
+            data: session,
+            metadata: {
+                name: nameSelector(state),
+                attributes: {
+                    user: userSelector(state).name
+                }
+            }
+        };
+        const id = idSelector(state);
+        return (id ? updateResource({
+            ...resource,
+            id
+        }) : createResource(resource)).switchMap(rid => {
+            return Rx.Observable.of(
+                userSessionSaved(rid, session)
+            );
+        })
+            .let(wrapStartStop(
+                loading(true, 'userSessionSaving'),
+                loading(false, 'userSessionSaving'),
+                ({ status, data }, stream$) => {
+                    if (status === 404 && isString(data) && data.indexOf('Resource Category not found') > -1) {
+                        return createCategory('USERSESSION').switchMap(() => stream$.skip(1))
+                            .catch(() => Rx.Observable.of(error({
+                                title: 'userSession.saveErrorNotification.titleContext',
+                                message: 'userSession.saveErrorNotification.categoryError',
+                                position: "tc",
+                                autoDismiss: 5,
+                                values: {
+                                    categoryName: 'USERSESSION'
+                                }
+                            })));
+                    }
+                    return Rx.Observable.of(error({
+                        title: 'userSession.saveErrorNotification.titleContext',
+                        message: saveUserSessionErrorStatusToMessage(status),
+                        position: "tc",
+                        autoDismiss: 5,
+                        values: {
+                            data
+                        }
+                    }));
+                }
+            ));
+    });
+
+export const startStopUserSessionSaveEpic = (startAction, endAction, frequency) => (action$) => action$
```

Comment:
The same here. We are not forced to have unique names for these epicCreators, therefore we can simple call it something like `autoSaveEpicCreator` or `autoSaveSessionEpicCreator`

Reply:
fine